tests/unit/test_utils.py

```diff
@@ -260,3 +271,74 @@ def test_does_dss_pvc_exist(lightkube_client_side_effect, context_raised, expect
 
     with context_raised:
         assert does_dss_pvc_exist(mock_client) == expected_return
+
+
+@pytest.mark.parametrize(
+    "lightkube_client_side_effect, context_raised, expected_return",
+    [
+        # The namespace is found (lightkube_client.get() does not fail)
+        (None, does_not_raise(), True),
+        # The namespace is not found
+        (FakeApiError(404), does_not_raise(), False),
+        # Some other ApiError is raised, which we don't know how to handle
+        (FakeApiError(999), pytest.raises(ApiError), None),
+    ],
+)
+def test_does_namespace_exist(lightkube_client_side_effect, context_raised, expected_return):
+    """Test the does_namespace_exist function."""
+    mock_client = MagicMock()
+    mock_client.get.side_effect = lightkube_client_side_effect
+
+    with context_raised:
+        assert does_namespace_exist(mock_client, "namespace") == expected_return
+
+
+@pytest.mark.parametrize(
+    "return_value, context_raised, side_effect, error_message, logger_info_call_count",
+    [
+        # Namespace has been deleted when the function is called
+        (False, does_not_raise(), None, None, 1),
+        # Times out while waiting for namespace to be deleted.
+        (
+            True,
+            pytest.raises(TimeoutError),
+            None,
+            "Timed out waiting for namespace {namespace} to be deleted.",
+            2,
+        ),
+        # Raises an unexpected error.
+        (False, pytest.raises(FakeApiError), FakeApiError(999), "broken", 1),
+    ],
+)
+def test_wait_for_namespace_to_be_deleted(
+    mock_does_namespace_exist: MagicMock,
+    mock_logger: MagicMock,
+    return_value: bool,
+    context_raised,
+    side_effect: Exception,
+    error_message: str,
+    logger_info_call_count: int,
+) -> None:
+    """
+    Test the three cases for wait_for_namespace_to_be_deleted function.
+    """
+    mock_client_instance = MagicMock()
+    namespace = "test-namespace"
+    mock_does_namespace_exist.return_value = return_value
+    mock_does_namespace_exist.side_effect = side_effect
+
+    # Call the function to test
+    with context_raised as exc_info:
+        wait_for_namespace_to_be_deleted(
+            mock_client_instance,
+            namespace=namespace,
+            timeout_seconds=2,
+            interval_seconds=1,
+        )
+
+    # Assertions
+    mock_logger.info.assert_called_with(f"Waiting for namespace {namespace} to be deleted...")
+    assert mock_logger.info.call_count == logger_info_call_count
+    if error_message:
+        error_message = error_message.replace("{namespace}", namespace)
```

Comment:
I think you want to use [`.format`](https://docs.python.org/3.10/library/stdtypes.html#str.format) here

Reply:
Resolving since this part is removed.